crates/bevy_ui/src/focus.rs

```diff
@@ -143,24 +143,68 @@ pub struct NodeQuery {
     target_camera: Option<&'static TargetCamera>,
 }
 
+/// Stores the ui cursor coordinates for cameras, used internally for ui focus management.
+///
+/// For optimal ui latency the contents should be updated in [`bevy_app::PreUpdate`], after [`bevy_input::InputSystem`] and before [`ui_focus_system`].
+/// for window-based cameras this is set automatically in [`set_camera_window_cursor_position`].
+#[derive(Component, Default)]
+pub struct CameraCursorPosition(pub Option<Vec2>);
+
+/// Set the cursor position for cameras that render to a Window.
+pub fn set_camera_window_cursor_position(
+    mut commands: Commands,
+    mut camera_query: Query<(Entity, &Camera, Option<&mut CameraCursorPosition>)>,
+    primary_window: Query<Entity, With<PrimaryWindow>>,
+    windows: Query<&Window>,
+    touches_input: Res<Touches>,
+    ui_scale: Res<UiScale>,
+) {
+    let primary_window = primary_window.iter().next();
+    for (camera_entity, camera, maybe_cursor) in &mut camera_query {
+        let maybe_window_ref = match camera.target.normalize(primary_window) {
+            Some(NormalizedRenderTarget::Window(w)) => Some(w),
+            _ => None,
+        };
+
+        let cursor_position = maybe_window_ref
+            .and_then(|w| windows.get(w.entity()).ok())
+            .and_then(Window::cursor_position)
+            .or_else(|| touches_input.first_pressed_position())
```

Comment:
just noting there's an assumption here that with touch devices there's only one window. not sure that's guaranteed.

Reply:
(there are also up to 12 pointers at a time)